ast/compile.go

```diff
@@ -1305,6 +1312,44 @@ func (c *Compiler) checkDuplicateImports() {
 	}
 }
 
+func (c *Compiler) checkKeywordOverrides() {
+	for _, name := range c.sorted {
+		mod := c.Modules[name]
+		errs := checkKeywordOverrides(mod, c.strict)
+		for _, err := range errs {
+			c.err(err)
+		}
+	}
+}
+
+func checkKeywordOverrides(node interface{}, strict bool) Errors {
+	if !strict {
+		return nil
+	}
+
+	errors := Errors{}
+
+	WalkRules(node, func(rule *Rule) bool {
+		name := rule.Head.Name.String()
+		if _, override := strictKeywords[name]; override {
+			errors = append(errors, NewError(CompileErr, rule.Location, "rule overrides %s", name))
+		}
+		return false
+	})
+
+	WalkExprs(node, func(expr *Expr) bool {
+		if expr.IsAssignment() {
+			name := expr.Operand(0).String()
+			if _, override := strictKeywords[name]; override {
+				errors = append(errors, NewError(CompileErr, expr.Location, "assignment overrides %s", name))
```

Comment:
Just tried defining `f(input) { input > 2 }` and it gives:

```
rego_compile_error: args must not shadow input (use a different variable name)
```
Do we want something similar here? (that is https://github.com/open-policy-agent/opa/blob/1f98d491fcf16ac6972d21130bd2a3e202007841/ast/compile.go#L3252-L3258)

Looks like we've re-used RootDocument there (https://github.com/open-policy-agent/opa/blob/a75b74db159a506305848ae6ad0f2510185826dd/ast/policy.go#L74-L77), which at least for now is the almost the same thing as `strictKeywords`

Reply:
Nice! I'll reuse that 👍 .